src/Chapter1.hs

```diff
@@ -653,7 +661,11 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n
+    | div (abs n) 10 == 0 = abs n
+    | div (abs n) 10 > 9 = firstDigit(div (abs n) 10)
+    | otherwise = div (abs n) 10
```

Comment:
Your solution is correct :100: But you can simplify a few checks here to make the implementation simpler. For example, if you check that `abs n` is less than 10, you can return `abs n` immediately. Otherwise, call a function recursively.

Also, you can see that `abs n` is reused several times in this function. This usually means that you can move it into the `where` case and reuse everywhere 👍🏼